DataManagementSystem/Client/ConsistencyChecks.py

```diff
@@ -0,0 +1,836 @@
+    """ Helper function - get files from the TS
+    """
+
+    selectDict = { 'TransformationID': self.prod}
+    if self._lfns:
+      selectDict['LFN'] = self._lfns
+    elif self.runStatus and self.fromProd:
+      res = self.transClient.getTransformationRuns( {'TransformationID': self.fromProd, 'Status':self.runStatus} )
+      if not res['OK']:
+        gLogger.error( "Failed to get runs for transformation %d" % self.prod )
+      else:
+        if res['Value']:
+          self.runsList.extend( [run['RunNumber'] for run in res['Value'] if run['RunNumber'] not in self.runsList] )
+          gLogger.always( "%d runs selected" % len( res['Value'] ) )
+        elif not self.runsList:
+          gLogger.always( "No runs selected, check completed" )
+          DIRAC.exit( 0 )
+    if not self._lfns and self.runsList:
+      selectDict['RunNumber'] = self.runsList
+
+    res = self.transClient.getTransformation( self.prod )
+    if not res['OK']:
+      gLogger.error( "Failed to find transformation %s" % self.prod )
+      return [], [], []
+    status = res['Value']['Status']
+    if status not in ( 'Active', 'Stopped', 'Completed', 'Idle' ):
+      gLogger.always( "Transformation %s in status %s, will not check if files are processed" % ( self.prod, status ) )
+      processedLFNs = []
+      nonProcessedLFNs = []
+      nonProcessedStatuses = []
+      if self._lfns:
+        processedLFNs = self._lfns
+    else:
+      res = self.transClient.getTransformationFiles( selectDict )
+      if not res['OK']:
+        gLogger.error( "Failed to get files for transformation %d" % self.prod, res['Message'] )
+        return [], [], []
+      else:
+        processedLFNs = [item['LFN'] for item in res['Value'] if item['Status'] == 'Processed']
+        nonProcessedLFNs = [item['LFN'] for item in res['Value'] if item['Status'] != 'Processed']
+        nonProcessedStatuses = list( set( [item['Status'] for item in res['Value'] if item['Status'] != 'Processed'] ) )
+
+    return processedLFNs, nonProcessedLFNs, nonProcessedStatuses
+
+  ################################################################################
+
+  def __write( self, text ):
+    if self.interactive:
+      sys.stdout.write( text )
+      sys.stdout.flush()
+
+  ################################################################################
+
+  def _selectByFileType( self, lfnDict, fileTypes = None, fileTypesExcluded = None ):
+    """ Select only those files from the values of lfnDict that have a certain type
+    """
+    if not lfnDict:
+      return {}
+    if not fileTypes:
+      fileTypes = self.fileType
+    if not fileTypesExcluded:
+      fileTypesExcluded = self.fileTypesExcluded
+    else:
+      fileTypesExcluded += [ft for ft in self.fileTypesExcluded if ft not in fileTypesExcluded]
+    # lfnDict is a dictionary of dictionaries including the metadata, create a deep copy to get modified
+    ancDict = copy.deepcopy( lfnDict )
+    if fileTypes == ['']:
```

Comment:
why would you get this as input ? Maybe it would be good to correct the code that generates this awful thing :-)